src/Float/Extra.elm

```diff
@@ -0,0 +1,377 @@
+module Float.Extra exposing
+    ( aboutEqual
+    , toFixedDecimalPlaces, toFixedSignificantDigits, boundaryValuesAsUnicode
+    , range
+    )
+
+{-|
+
+
+# Equality
+
+@docs aboutEqual
+
+
+# Formatting Floats
+
+@docs toFixedDecimalPlaces, toFixedSignificantDigits, boundaryValuesAsUnicode
+
+
+# Ranges
+
+@docs range
+
+-}
+
+-- toFixedDecimalDigits implementation
+
+
+zeroes : Float -> String
+zeroes v =
+    String.repeat (floor v) "0"
+
+
+adjustDecimalPlace : Float -> Float -> Float
+adjustDecimalPlace x magnitude =
+    if magnitude < 0 then
+        x * 10 ^ -magnitude
+
+    else
+        x / 10 ^ magnitude
+
+
+{-| As specified in <https://tc39.es/ecma402/#sec-torawprecision>
+-}
+toRawPrecision : Float -> Float -> String
+toRawPrecision x precision =
+    let
+        ( significantBaseString, decimalPointIndex ) =
+            toSignificantBase precision x
+    in
+    if decimalPointIndex >= precision then
+        significantBaseString ++ zeroes (decimalPointIndex - precision + 1)
+
+    else if decimalPointIndex == precision - 1 then
+        significantBaseString
+
+    else
+        let
+            candidateString =
+                if decimalPointIndex >= 0 then
+                    String.slice 0 (floor (decimalPointIndex + 1)) significantBaseString ++ "." ++ String.dropLeft (floor (decimalPointIndex + 1)) significantBaseString
```

Comment:
:nail_care: extract `(floor (decimalPointIndex + 1))` into a local `let` so it's clearer you're slicing at the same point

Reply:
ugh, actually in this case that requires introducing yet another nested lambda (due to elm-review), so probably not worth it imo...